packages/material-ui/src/Typography/Typography.js

```diff
@@ -5,37 +5,32 @@ import { componentPropType } from '@material-ui/utils';
 import withStyles from '../styles/withStyles';
 import { capitalize } from '../utils/helpers';
 
+const allVariants = [
+  'h1',
+  'h2',
+  'h3',
+  'h4',
+  'h5',
+  'h6',
+  'subtitle1',
+  'subtitle2',
+  'body1',
+  'body2',
+  'caption',
+  'button',
+  'overline',
+];
+
+/* styles of variants in theme.typography */
+const typographyStyles = theme =>
+  allVariants.reduce((newObj, x) => Object.assign(newObj, { [x]: theme.typography[x] }), {});
```

Comment:
nm, this breaks docs tests as the removed comments are be used for docs

Reply:
Why was this change even required? I mean it is less code but I wouldn't mind the verbosity here.

Reply:
https://github.com/mui-org/material-ui/pull/14573#discussion_r257718523